spec/datadog/appsec/waf_spec.rb

```diff
@@ -202,6 +202,41 @@
         end
       end
     end
+
+    context 'stress testing' do
+      let(:run_count) { 500 }
+      let(:thread_count) { 200 }
+
+      it 'creates a context in each thread' do
+        handle
+
+        result = { ok: 0, match: 0, err_internal: 0, err_invalid_object: 0, err_invalid_argument: 0 }
+        start_barrier = Barrier.new(thread_count)
+        res_mutex = Mutex.new
+        threads = []
+
+        thread_count.times do
+          threads << Thread.new do
+            local_context = Datadog::AppSec::WAF::Context.new(handle)
```

Comment:
I suggest simplify naming here, there is no non-local context here

```suggestion
            context = Datadog::AppSec::WAF::Context.new(handle)
```

Reply:
Each test case must be in isolation (ideally). If you don't need something defined before you can overwrite it.

Reply:
Fixed in 7957a106fa17cca4ffad1352764789c13179051a